libs/core/synchronization/include/hpx/synchronization/condition_variable.hpp

```diff
@@ -181,7 +540,22 @@ namespace hpx {
         hpx::util::cache_aligned_data_derived<data_type> data_;
     };
 
-    ///////////////////////////////////////////////////////////////////////////
+    /**
+     * \brief The \a condition_variable_any class is a generalization of
+     * \a hpx::condition_variable. Whereas \a hpx::condition_variable works
+     * only on \a std::unique_lock<std::mutex>, \a acondition_variable_any can
+     * operate on any lock that meets the \a BasicLockable requirements.
+     *
+     * See \a hpx::condition_variable for the description of the semantics of
+     * condition variables.
+     * The class \a hpx::condition_variable_any is a \a StandardLayoutType. It
+     * is not \a CopyConstructible, \a MoveConstructible, \a CopyAssignable,
+     * or \a MoveAssignable.
+     *
+     * FIXME(bhumit->hkaiser) true for std, what about hpx?
+     * If the lock is \a std::unique_lock<std::mutex>,
+     * \a hpx::condition_variable may provide better performance.
```

Comment:
@hkaiser is this true for hpx as well?

Reply:
I don't think so. HPX does not implement any special optimizations for `condition_variable` (which is the only reason why there are two different classes in the first place). Implementation wise in HPX both, `condition_variable` and `condition_variable_any` are almost identical.